core/src/main/java/brooklyn/entity/basic/BrooklynTaskTags.java

```diff
@@ -346,6 +347,26 @@ public static boolean isInEffectorTask(Task<?> task, @Nullable Entity entity, @N
         return false;
     }
 
+    @Beta
+    public static Task<?> getEffectorParentTask(Task<?> child, Effector<?> effector) {
```

Comment:
rename as `getClosestEffectorTask()` ?  it's not necessarily the parent.  `closest` is used in the jquery sense.

might be useful eventually to refactor to use predicates, but probably not yet...

and add a one-line javadoc.